addons/stock/models/stock_rule.py

```diff
@@ -483,10 +483,12 @@ def _merge_domain(self, values, rule, group_id):
 
     @api.model
     def _get_moves_to_assign_domain(self):
+        reservation_domain = ['|', '|', ('picking_type_id.reservation_method', '=', 'at_confirm'),
+                              ('picking_id.picking_type_id.reservation_method', '=', 'at_confirm'),
+                              ('reservation_date', '<=', fields.Datetime.now())]
```

Comment:
We really need to work on making the picking_type_id on stock move computed. I don't think we should do it in this task but in a clean separate one because this kind of domains (and it's unfortunately not the first time in Stock) is awful :D
(cc @amoyaux @ryv-odoo)

Reply:
Agreed. I wasn't sure if there was a reason why it's set up the way it currently is (if it's specially used to distinguish manufacturing moves or not). So for now I leave this as is?